### PR TITLE
Schema implementation

### PR DESCRIPTION
**Pull request by rafaelmon14**

### CODE REVIEW

File: lib/models/knowledge-areas.dart [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  String name;
+
+  /// full description of the knowledge area
+  String description;
+
+  /// number of questions blocks this knowledge area contains
+  int numberOfQuestionsBlocks;
+
+  /// list of the content the knowledge area includes
+  Map<String, int> contentSpecifications = {};
```
Comment (baesparza) [inline lib/models/knowledge-areas.dart:17]: Este no seria un Map<String,int> sino solo un List<string>, osea un arreglo

File: lib/models/questions.dart [hunk 2]
```diff
@@ -0,0 +1,162 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+/// Docs at https://github.com/UTPL-Rank/UTPL-Rank-Models/blob/master/Cloud_Firestore_Data_Model.md#academic-areas
+class Questions {
+
+  /// identifier of the question
+  String id;
+
+  /// how hard is the question
+  dynamic dificulty;
```
Comment (baesparza) [inline lib/models/questions.dart:10]: this is a string

File: lib/models/questions.dart [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  }
+}
+
+
+class _QuestionSequence {
+
+  /// next question reference
+  /// 
+  /// @Optional
+  dynamic next;
```
Comment (baesparza) [inline lib/models/questions.dart:75]: this is a document reference

File: lib/models/questions.dart [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+  /// previous question reference
+  /// 
+  /// @Optional
+  dynamic prev;
```
Comment (baesparza) [inline lib/models/questions.dart:80]: this is a reference

File: lib/models/questions.dart (hunk 2 above)
Comment (baesparza) [inline lib/models/questions.dart:10]: this is a string

File: lib/models/questions.dart [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+  /// default message to be show in case of wrong answer
+  /// 
+  /// @Optional
+  String defaultWrongAnswerMessage;
+
+  /// Used for question sequencing, when a question combes before or after another question
+  _QuestionSequence questionSequence;
+
+  /// image part of the question
```
Comment (baesparza) [inline lib/models/questions.dart:32]: add the optional tag

File: lib/models/block-responses.dart [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+/// Docs at https://github.com/UTPL-Rank/UTPL-Rank-Models/blob/master/Cloud_Firestore_Data_Model.md#academic-areas
+class BlockResponses {
+  /// identifier of the response
+  String id;
+
+  /// reference to the user document
+  dynamic userReference;
```
Comment (baesparza) [inline lib/models/block-responses.dart:9]: this should be a documentReference

File: lib/models/users.dart [hunk 7]
```diff
@@ -0,0 +1,39 @@
+/// Docs at https://github.com/UTPL-Rank/UTPL-Rank-Models/blob/master/Cloud_Firestore_Data_Model.md#academic-areas
+class Users {
```
Comment (baesparza) [inline lib/models/users.dart:2]: Rename to User

File: lib/models/questions.dart [hunk 8]
```diff
@@ -0,0 +1,162 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+/// Docs at https://github.com/UTPL-Rank/UTPL-Rank-Models/blob/master/Cloud_Firestore_Data_Model.md#academic-areas
+class Questions {
```
Comment (baesparza) [inline lib/models/questions.dart:4]: rename to question

File: lib/models/knowledge-areas.dart [hunk 9]
```diff
@@ -0,0 +1,57 @@
+/// Docs at https://github.com/UTPL-Rank/UTPL-Rank-Models/blob/master/Cloud_Firestore_Data_Model.md#academic-areas
+class KnowledgeAreas {
```
Comment (baesparza) [inline lib/models/knowledge-areas.dart:2]: reaname KnoeledgeArea

File: lib/models/games-played.dart [hunk 10]
```diff
@@ -0,0 +1,94 @@
+
+import 'package:cloud_firestore/cloud_firestore.dart';
+/// Docs at https://github.com/UTPL-Rank/UTPL-Rank-Models/blob/master/Cloud_Firestore_Data_Model.md#academic-areas
+class GamesPlayed{
```
Comment (baesparza) [inline lib/models/games-played.dart:4]: rename to GamePlayed

File: lib/models/block-responses.dart [hunk 11]
```diff
@@ -0,0 +1,35 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+/// Docs at https://github.com/UTPL-Rank/UTPL-Rank-Models/blob/master/Cloud_Firestore_Data_Model.md#academic-areas
+class BlockResponses {
```
Comment (baesparza) [inline lib/models/block-responses.dart:4]: rename to BlockResponse

File: lib/models/academic-degree.dart [hunk 12]
```diff
@@ -0,0 +1,154 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+/// Docs at https://github.com/UTPL-Rank/UTPL-Rank-Models/blob/master/Cloud_Firestore_Data_Model.md#academic-areas
+class AcademicDegrees {
```
Comment (baesparza) [inline lib/models/academic-degree.dart:3]: Rename to AcademicDegree

**Comment by rafaelmon14** [comment]
done

**Pull request closed without merging**